### PR TITLE
Add check for 64bit builtin atomics

### PR DESCRIPTION
**Pull request by kraj**
Compilers for some architectures e.g. RISCV32 do not have 64bit atomics therefore, its important to deduce that and use it to pass right flags to compiler to use them, e.g. h2o module

Fixes
nection.c.o): in function `h2o_create_connection': | /usr/src/debug/netdata/1.47.1/src/web/server/h2o/libh2o/include/h2o.h:1917:(.text.create_conn+0x5e): undefined reference to `__sync_add_and_fetch_8' | collect2: error: ld returned 1 exit status

### CODE REVIEW

File: CMakeLists.txt
Comment (Ferroin) [inline CMakeLists.txt:541]: - This should be down with the other H2O related compliation stuff (around line 1650) so that it’s clear that it’s tied to H2O itself, not to our own H2O support.
- This needs to be added using `target_compile_options()` or better yet `target_compile_definitions()` for the `h2o` target so that it only affects that target and things which depend on that target, as well as being properly applied regardless of whether the thing being compiled is C or C++. `${CMAKE_C_FLAGS}` only affects C compiler invocations, and it will bleed over into essentially every external project we use as well which is undesirable for multiple reasons.

**Review by Ferroin — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (kraj) [reply CMakeLists.txt]: @Ferroin thanks for review, I have addressed your feedback  and updated the PR

**Review by Ferroin — APPROVED** [review]

**Comment by Ferroin** [comment]
@netdata/agent Could we get a second review on this? It looks correct from my perspective, but I may be missing something so I’d appreciate confirmation at least that nothing is broken by this.

**Comment by Ferroin** [comment]
@kraj Thanks for the contribution!

**Comment by Ferroin** [comment]
@stelfrag This one should probably go in the next patch release.

**Pull request merged**
